datastream-server/src/main/java/com/linkedin/datastream/server/DatastreamTaskImpl.java

```diff
@@ -479,6 +479,8 @@ public String toString() {
       } catch (NumberFormatException e) {
         LOG.error(e.getMessage());
       }
+    } else { // this will be called for low level kafka connector.
```

Comment:
This should not be in else block. Check line 472.

`// When DatastreamTaskImpl is created using constructor that passes _partitions, _partitionsV2 is also populated.`

Reply:
my bad !! makes sense.